assets/questions/unknownAlphabet.js

```diff
@@ -0,0 +1,326 @@
+
+/**
+ * @fileoverview Question data for Unknown Alphabet.
+ */
+
+globalData.questions['unknownAlphabet'] = {  // eslint-disable-line dot-notation
+  title: 'Unknown Alphabet',
+  starterCode: {
+    python:
+`def findDictionary(words):
+    return ""`
+  },
+  auxiliaryCode: {
+    python:
+`
+class AuxiliaryCode(object):
+  @classmethod
+  def failsOnNoSolution(cls, words):
+    """
+      edges stores the edges that need to build for the alphabet graph
+      edges[ch] is a list of letters for letter ch whose ranks are bigger than
+      ch's rank.
+      degree stores the degree of each letter.
+      For each letter ch, there are degree[ch] letters that have an edge 
+      connecting to ch.
+    """
+    edges = {}
+    degree = {}
+    for word in words:
+      for ch in word:
+        ch = ch.lower()
+        if(not edges.has_key(ch)):
+          edges[ch] = []
+          degree[ch] = 0
+    for wordIndex in range(len(words) - 1):
+      for letterIndex in range(min(len(words[wordIndex]), len(words[wordIndex + 1]))):
+        letterOne = words[wordIndex][letterIndex].lower()
+        letterTwo = words[wordIndex + 1][letterIndex].lower()
+        if(letterOne != letterTwo):
+          edges[letterOne].append(letterTwo)
+          degree[letterTwo] = degree[letterTwo] + 1
+          break
+    answer  = ""
+    hasZeroDegreeLetter = True
+    while hasZeroDegreeLetter :
+      hasZeroDegreeLetter = False
+      for key in degree:
+        if degree[key] == 0:
+          hasZeroDegreeLetter = True
+          answer = answer + key
+          for nextLetter in edges[key]:
+            degree[nextLetter] = degree[nextLetter] - 1
+          del degree[key]
+      if not hasZeroDegreeLetter:
+        break
+    return answer
+    
+    
+  @classmethod
+  def failsOnUppercase(cls, words):
+    edges = {}
+    degree = {}
+    for word in words:
+      for ch in word:
+        if(not edges.has_key(ch)):
+          edges[ch] = []
+          degree[ch] = 0
+    for wordIndex in range(len(words) - 1):
+      for letterIndex in range(min(len(words[wordIndex]), len(words[wordIndex + 1]))):
+        letterOne = words[wordIndex][letterIndex]
+        letterTwo = words[wordIndex + 1][letterIndex]
+        if letterOne != letterTwo:
+          edges[letterOne].append(letterTwo)
+          degree[letterTwo] = degree[letterTwo] + 1
+          break
+    answer  = ""
+    hasZeroDegreeLetter = True
+    while hasZeroDegreeLetter :
+      hasZeroDegreeLetter = False
+      for key in degree:
+        if degree[key] == 0:
+          hasZeroDegreeLetter = True
+          answer = answer + key
+          for nextLetter in edges[key]:
+            degree[nextLetter] = degree[nextLetter] - 1
+          del degree[key]
+      if not hasZeroDegreeLetter:
+        break
+    if degree:
+      return ""
+    return answer
+`
+ 
+  },
+  tasks: [{
+    instructions: [
+      [
+        'Given a dictionary (a list of words in lexicographic order) of all',
```

Comment:
If we're providing a dictionary and we want an alphabet, why is the method called findDictionary and not findAlphabet?